Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Select.hs

```diff
@@ -73,8 +73,8 @@ handler merchant sReq estimate = do
           driverSelectedFare = Nothing,
           customerExtraFee = sReq.customerExtraFee
         }
-  searchTry <- buildSearchTry merchantId searchReq.id estimate sReq.customerExtraFee sReq searchReq.estimatedDistance searchReq.estimatedDuration
   let estimateFare = fareSum fareParams
```

Comment:
Lets compare this new fare with older estimatedFare if it is matching then only go ahead else fail.

Reply:
This way it will handle search happened before night charges applies and retry happened just after that.